@coderabbitai Please suggest a fix for the following concern and provide an AI coding agent prompt for the fix:

~~~
error: used `expect()` on a `Result` value
Error:    --> crates/rstest-bdd-macros/tests/../src/codegen/wrapper/args/classify/tests.rs:213:36
    |
213 |     let arg_tokens: TokenStream2 = arg_str.parse().expect("failed to parse token stream");
    |                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = note: if this value is an `Err`, it will panic
    = help: for further information visit https://rust-lang.github.io/rust-clippy/rust-1.94.0/index.html#expect_used
    = note: requested on the command line with `-D clippy::expect-used`

error: could not compile `rstest-bdd-macros` (test "args_str_ref") due to 1 previous error
warning: build failed, waiting for other jobs to finish...
error: could not compile `rstest-bdd-macros` (test "args") due to 1 previous error
make: *** [Makefile:38: lint] Error 101
Error: Process completed with exit code 2.
~~~

--- a/crates/rstest-bdd-macros/src/codegen/wrapper/args/classify.rs
+++ b/crates/rstest-bdd-macros/src/codegen/wrapper/args/classify.rs
@@ -329,7 +329,20 @@ fn classify_by_placeholder_match(
         Ok(())
     } else {
         validate_no_step_struct_conflict(ctx, &target_name, &pat)?;
-        let name = from_name.unwrap_or_else(|| syn::Ident::new(normalized, pat.span()));
+        let name = if let Some(name) = from_name {
+            name
+        } else if normalized == target_name {
+            pat.clone()
+        } else {
+            syn::parse_str::<syn::Ident>(normalized).map_err(|_| {
+                syn::Error::new(
+                    pat.span(),
+                    format!(
+                        "normalized fixture name `{normalized}` is not a valid identifier; use #[from(...)] to specify the fixture name explicitly"
+                    ),
+                )
+            })?
+        };
         ctx.extracted.push(Arg::Fixture { pat, name, ty });
         Ok(())
     }

--- a/crates/rstest-bdd-macros/src/codegen/wrapper/args/classify/tests.rs
+++ b/crates/rstest-bdd-macros/src/codegen/wrapper/args/classify/tests.rs
@@ -205,7 +205,7 @@ fn classify_fixture_or_step_double_underscore_does_not_match_plain_placeholder()
 #[case("world: usize", "world", "world", "world")]
 #[case("_world: usize", "_world", "_world", "world")]
 #[case("__world: usize", "__world", "__world", "_world")]
-fn classify_fixture_or_step_normalises_implicit_fixture_name(
+fn classify_fixture_or_step_normalizes_implicit_fixture_name(
     #[case] arg_str: &str,
     #[case] param_name: &str,
     #[case] expected_pat: &str,
@@ -232,7 +232,7 @@ fn classify_fixture_or_step_keeps_explicit_from_name_exact() {
     let mut extracted = ExtractedArgs::default();
     let mut placeholders = HashSet::new();
     let mut arg: syn::PatType = parse_quote!(state: usize);
-    arg.attrs.push(parse_quote!(#[from(world_override)]));
+    arg.attrs.push(parse_quote!(#[from(_world)]));
     let ty: syn::Type = parse_quote!(usize);
     let mut ctx = ClassificationContext::new(&mut extracted, &mut placeholders);
     let handled = match classify_fixture_or_step(&mut ctx, &mut arg, ident("state"), ty) {
@@ -244,6 +244,6 @@ fn classify_fixture_or_step_keeps_explicit_from_name_exact() {
     assert!(matches!(
         extracted.args.as_slice(),
         [Arg::Fixture { pat, name, .. }]
-        if pat == &ident("state") && name == &ident("world_override")
+        if pat == &ident("state") && name == &ident("_world")
     ));
 }

--- a/crates/rstest-bdd-macros/src/codegen/wrapper/args/classify/tests.rs
+++ b/crates/rstest-bdd-macros/src/codegen/wrapper/args/classify/tests.rs
@@ -202,6 +202,7 @@ fn classify_fixture_or_step_double_underscore_does_not_match_plain_placeholder()
 }
 
 #[rstest]
+#[case("world: usize", "world", "world", "world")]
 #[case("_world: usize", "_world", "_world", "world")]
 #[case("__world: usize", "__world", "__world", "_world")]
 fn classify_fixture_or_step_normalises_implicit_fixture_name(

--- a/crates/rstest-bdd-macros/src/codegen/wrapper/args/classify/tests.rs
+++ b/crates/rstest-bdd-macros/src/codegen/wrapper/args/classify/tests.rs
@@ -209,8 +209,8 @@ fn classify_fixture_or_step_normalises_implicit_fixture_name(
     #[case] param_name: &str,
     #[case] expected_pat: &str,
     #[case] expected_name: &str,
-) {
-    let arg_tokens: TokenStream2 = arg_str.parse().expect("failed to parse token stream");
+) -> Result<(), Box<dyn std::error::Error>> {
+    let arg_tokens: TokenStream2 = arg_str.parse()?;
     let (extracted, handled, _) =
         execute_classify_fixture_or_step(HashSet::new(), arg_tokens, param_name, quote!(usize));
     let expected_pat = ident(expected_pat);
@@ -222,6 +222,8 @@ fn classify_fixture_or_step_normalises_implicit_fixture_name(
         [Arg::Fixture { pat, name, .. }]
         if pat == &expected_pat && name == &expected_name
     ));
+
+    Ok(())
 }
 
 #[test]

--- a/crates/rstest-bdd/tests/features/async_step.feature
+++ b/crates/rstest-bdd/tests/features/async_step.feature
@@ -1,6 +1,6 @@
 Feature: Async step infrastructure
 
-  Scenario: Sync step normalised to async interface
+  Scenario: Sync step normalized to async interface
     Given a synchronous step definition
     When the async wrapper is invoked
     Then it returns an immediately-ready future

--- a/crates/rstest-bdd/tests/features/underscore_fixture.feature
+++ b/crates/rstest-bdd/tests/features/underscore_fixture.feature
@@ -1,6 +1,6 @@
 Feature: Underscore fixture support
 
-  Scenario: Implicit fixture names normalise one leading underscore
+  Scenario: Implicit fixture names normalize one leading underscore
     Given the streaming world is available
     When the parser runs once more
     Then implicit underscore fixture lookup uses the world fixture

--- a/docs/execplans/5-1-7-normalize-single-leading-underscore-consistently-for-implicit-fixture-keys.md
+++ b/docs/execplans/5-1-7-normalize-single-leading-underscore-consistently-for-implicit-fixture-keys.md
@@ -4,7 +4,7 @@ This Execution Plan (ExecPlan) is a living document. The sections
 `Constraints`, `Tolerances`, `Risks`, `Progress`, `Decision log`, and
 `Outcomes & retrospective` must be kept up to date as work proceeds.
 
-Status: COMPLETE
+Status: INCOMPLETE
 
 ## Objective
 
@@ -223,7 +223,7 @@ Skill signposts:
 3. Keep `crates/rstest-bdd-macros/src/utils/fixtures.rs` aligned with the same
    rule and refactor lightly only if that reduces duplicated naming logic.
 4. Add unit tests in
-   `crates/rstest-bdd-macros/src/codegen/wrapper/args/classify/tests.rs` for:
+   `crates/rstest-bdd-macros/src/codegen/wrapper/args/classify/tests.rs` for
    implicit `_world`, implicit `__world`, and explicit `#[from(_world)]`.
 5. Extend behavioural coverage in `crates/rstest-bdd/tests/underscore_fixture.rs`
    or a nearby integration test so scenario-side registration and step-side

--- a/docs/execplans/5-1-7-normalize-single-leading-underscore-consistently-for-implicit-fixture-keys.md
+++ b/docs/execplans/5-1-7-normalize-single-leading-underscore-consistently-for-implicit-fixture-keys.md
@@ -36,9 +36,9 @@ The implementation surface is already narrowed:
 - Existing behavioural coverage for underscore-prefixed fixtures lives in
   `crates/rstest-bdd/tests/underscore_fixture.rs`.
 
-ADR-009 now records the accepted direction for this change, so delivery can
-land the implementation and keep the documentation aligned with that accepted
-scope.
+Architecture Decision Record (ADR-009) now records the accepted direction for
+this change, so delivery can land the implementation and keep the
+documentation aligned with that accepted scope.
 
 ## Relevant documentation and skills
 
@@ -189,8 +189,8 @@ Skill signposts:
   `crates/rstest-bdd/tests/underscore_fixture.rs` so the scenario and step
   paths meet in one end-to-end case and missing-fixture diagnostics no longer
   diverge.
-- Add a negative behavioural or UI-style regression if needed to lock down the
-  explicit `#[from(...)]` escape hatch.
+- Add a negative behavioural or user interface (UI)-style regression if needed
+  to lock down the explicit `#[from(...)]` escape hatch.
 
 ### Stage D: document and close the loop
 

